__tests__/app.test.js

```diff
@@ -334,3 +334,53 @@ describe("/api/articles/:article_id/comments", () => {
         });
     });
 });
+
+describe("/api/users", () => {
+    describe("GET", () => {
+        test("Returns 'Status: 200' with user objects contained in an array", () => {
+            return request(app).get('/api/users').expect(200)
+            .then(({ body }) => {
+                const users = body.users;
+                expect(Array.isArray(users)).toBe(true);
+            });
+        });
+        test("Array of users contains ALL users", () => {
+            return request(app).get('/api/users').expect(200)
+            .then(({ body }) => {
+                const users = body.users;
+                expect(users.length).toBe(4);
+            });
+        });
+        test("Each user object contains the keys 'username', 'name' and 'avatar_url'", () => {
+            return request(app).get('/api/users').expect(200)
+            .then(({ body }) => {
+                const users = body.users;
+                users.forEach((user) => {
+                    expect(user).toEqual(
+                        expect.objectContaining({
+                            username: expect.any(String),
+                            name: expect.any(String),
+                            avatar_url: expect.any(String)     
+                        })
+                    )
+                });
+            });
+        });
+        test("User object keys have the correct values", () => {
```

Comment:
This is really nice and thorough testing, but I'd probably say that it could be condensed into fewer tests. For example, where you're expecting any string in the test above would be sufficient rather than checking the actual values. You could probably also check the length of the array in the first test where you're checking for the array being returned.

Reply:
Thanks Rose, will refactor!